ui/components/app/collectible-details/collectible-details.js

```diff
@@ -0,0 +1,215 @@
+            >
+              {name}
+            </Typography>
+            <Typography
+              color={COLORS.UI3}
+              variant={TYPOGRAPHY.H5}
+              boxProps={{ marginTop: 2, marginBottom: 3 }}
+              overflowWrap={OVERFLOW_WRAP.BREAK_WORD}
+            >
+              {`#${tokenId}`}
+            </Typography>
+            <Typography
+              color={COLORS.BLACK}
+              variant={TYPOGRAPHY.H6}
+              fontWeight={FONT_WEIGHT.BOLD}
+              className="collectible-details__description"
+            >
+              {t('description')}
+            </Typography>
+            <Typography color={COLORS.UI3} variant={TYPOGRAPHY.H6}>
+              {description}
+            </Typography>
+          </Box>
+        </div>
+        <Box margin={4}>
+          <Box display={DISPLAY.FLEX} flexDirection={FLEX_DIRECTION.ROW}>
+            <Typography
+              color={COLORS.BLACK}
+              variant={TYPOGRAPHY.H6}
+              fontWeight={FONT_WEIGHT.BOLD}
+              boxProps={{ marginBottom: 3, width: BLOCK_SIZES.ONE_FOURTH }}
+            >
+              {t('source')}
+            </Typography>
+            <Typography
+              color={COLORS.PRIMARY1}
+              variant={TYPOGRAPHY.H6}
+              boxProps={{ marginBottom: 3, width: BLOCK_SIZES.THREE_FOURTHS }}
+              overflowWrap={OVERFLOW_WRAP.BREAK_WORD}
+            >
+              <a
+                target="_blank"
+                href={collectibleImageURL}
+                rel="noopener noreferrer"
+                className="collectible-details__image-link"
+              >
+                {image}
+              </a>
+            </Typography>
+          </Box>
+          <Box display={DISPLAY.FLEX} flexDirection={FLEX_DIRECTION.ROW}>
+            <Typography
+              color={COLORS.BLACK}
+              variant={TYPOGRAPHY.H6}
+              fontWeight={FONT_WEIGHT.BOLD}
+              boxProps={{ width: BLOCK_SIZES.ONE_FOURTH }}
+            >
+              {t('contractAddress')}
+            </Typography>
+            <Typography
+              color={COLORS.UI3}
+              variant={TYPOGRAPHY.H6}
+              overflowWrap={OVERFLOW_WRAP.BREAK_WORD}
+              boxProps={{ width: BLOCK_SIZES.THREE_FOURTHS }}
+            >
+              <a
+                target="_blank"
+                href={getTokenTrackerLink(
+                  address,
+                  currentNetwork,
+                  null,
+                  null,
+                  rpcPrefs,
+                )}
+                rel="noopener noreferrer"
+              >
+                {getEnvironmentType() === ENVIRONMENT_TYPE_POPUP
+                  ? shortenAddress(address)
+                  : address}
+              </a>
+            </Typography>
+          </Box>
+        </Box>
+      </Box>
+    </>
+  );
+}
+
+CollectibleDetails.propTypes = {
+  collectible: PropTypes.object,
```

Comment:
Is it possible for us to specify a shape for this object? Also should this be required?

Reply:
added improved typing here: https://github.com/MetaMask/metamask-extension/pull/12970/commits/5ec71ea225d74795b08c5206f38e420a4729d6a0